templates/types/streaming/nextjs/app/components/transform.ts

```diff
@@ -6,14 +6,60 @@ export const isValidMessageData = (rawData: JSONValue | undefined) => {
   return true;
 };
 
+const transformNodes = (nodes: any) => {
+  const SCORE_THRESHOLD = 0.5;
+  return nodes
+    .filter((node: any) => node.score > SCORE_THRESHOLD)
+    .sort((a: any, b: any) => b.score - a.score);
+};
+
 export const insertDataIntoMessages = (
   messages: Message[],
   data: JSONValue[] | undefined,
 ) => {
   if (!data) return messages;
   messages.forEach((message, i) => {
     const rawData = data[i];
-    if (isValidMessageData(rawData)) message.data = rawData;
+    if (isValidMessageData(rawData)) {
+      // If the message has nodes, transform them
+      if ((rawData as any).nodes?.length) {
+        message.data = {
```

Comment:
does this work with images? usually we have image URL in data

Reply:
maybe extract to a `transformMessageData` function that handles all these cases?